src/wp-admin/options.php

```diff
@@ -279,6 +279,23 @@
 			$_POST['gmt_offset']      = $_POST['timezone_string'];
 			$_POST['gmt_offset']      = preg_replace( '/UTC\+?/', '', $_POST['gmt_offset'] );
 			$_POST['timezone_string'] = '';
+		} elseif ( isset( $_POST['timezone_string'] ) && ! in_array( $_POST['timezone_string'], timezone_identifiers_list(), true ) ) {
```

Comment:
Mayybe?

I'm thinking of covering cases in which timezones have changed name, such as Kiev/Kyiv.

```suggestion
		} elseif ( isset( $_POST['timezone_string'] ) && ! in_array( $_POST['timezone_string'], timezone_identifiers_list( DateTimeZone::ALL_WITH_BC ), true ) ) {
```

Reply:
Great point! The form on `options-general.php` may have an earlier, now outdated timezone string added to the options list per [this commit](https://github.com/WordPress/wordpress-develop/commit/45da79e52e3252ddb8286ca0e6cd5db4522d874ca), where the `value` attribute is set to the same outdated value, not the potential "new" value. That means `$_POST['timezone_string']` _can_ be an outdated value, and would therefore be missed by this code. Using the `ALL_WITH_BC` isn't just a good idea, it's _requirer_ for backward compatibility.

Very nice catch Peter!